src/fifo/FifoController.h

```diff
@@ -32,19 +32,24 @@ class FifoController : public FifoControllerBase
     FifoController(uint32_t bufferSize, uint32_t threshold);
     virtual ~FifoController() = default;
 
-    // TODO review use atomics or memory barriers
```

Comment:
From what I could tell, the current use of memory barriers is adequate to keep mWriteCounter and mReadCounter consistent. However, I don't know if there could be problems in FifoControllerBase::getFullFramesAvailable(), when we access both variables. I can't tell if there is a scenario where two threads would get a different number of frames, that could be prevented by a use of std::memory_order_seq_cst instead. From here (https://en.cppreference.com/w/cpp/atomic/memory_order#Sequentially-consistent_ordering) :
"Sequential ordering may be necessary for multiple producer-multiple consumer situations where all consumers must observe the actions of all producers occurring in the same order."

How many producers/consumers can FifoBuffer have?

Reply:
Useful video:
https://youtu.be/ZQFzMfHIxng?t=3117

Reply:
This FIFO has only one reader and one writer.